lib/rubocop/cop/ezcater/rspec_require_browser_mock.rb

```diff
@@ -0,0 +1,56 @@
+module RuboCop
+  module Cop
+    module Ezcater
+      # Enforce use of `mock_ezcater_app`, `mock_chrome_browser` & `mock_custom_browser` helpers
+      # instead of mocking `Browser` or `EzBrowser` directly.
+      #
+      # @example
+      #
+      #   # good
+      #   mock_ezcater_app
+      #   mock_chrome_browser
+      #   mock_custom_browser
+      #   mock_ezcater_app(device: "My Device", version: "1.2.3", language: "en=US,en")
+      #   mock_chrome_browser(device: "My Device", version: "1.2.3", language: "en=US,en")
+      #   mock_custom_browser(user_agent: "My Custom Agent", language: "en=US,en"
+      #
+      #   # bad
+      #   allow(Browser).to receive...
+      #   allow(EzBrowser).to receive...
+      class RspecRequireBrowserMock < Cop
+        EZ_BROWSER_MSG = "Use the mocks provided by `BrowserHelpers` instead of mocking `allow(EzBrowser)`".freeze
+        BROWSER_MSG = "Use the mocks provided by `BrowserHelpers` instead of mocking `allow(Browser)`".freeze
+
+        def_node_matcher :browser_const?, <<~PATTERN
+          (const _ {:Browser, :EzBrowser})
+        PATTERN
+
+        def on_const(node)
+          return unless browser_const?(node)
+
+          # Walk to send node where method = :allow
+          match_node = node
+          match_node = match_node.parent while not_allow_send_node?(match_node.parent)
+          match_node = match_node.parent
+
+          # Validate send node, method = :allow; could have never been found
+          return unless allow_send_node?(match_node)
+
+          # Finish tree navigation to full line for highlighting
+          match_node = match_node.parent while match_node.parent
+          add_offense(match_node, :expression, RspecRequireBrowserMock.const_get("#{StringManipulator.underscore(node.source).upcase}_MSG"))
```

Comment:
I don't think you have to use a const here. You could just interpolate into a string.

Reply:
If keeping the constant implementation, then extract a helper method to get the message? Or do the node to browser class name conversion.

Reply:
switched to interpolation... I got fancy for some reason with this.